sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/batch_starter.py

```diff
@@ -815,12 +820,26 @@ def cadence_processing_event(session, events):
             continue
 
         logger.info(f"All required dependencies found for the dependency: {job_node}")
+        # The start date for the cadence job should be the earliest date of the
+        # upstream science dependency. For example, if the job is an ultra l2 map, then
+        # the start date should be the date of the first upstream pset.
+        # For cadence jobs, it is assumed there is at least one primary ScienceInput,
+        # typically only one.
+        primary_science_inputs = upstream_dependencies.get_science_inputs(
+            node["data_source"]
+        )
+        start_dates = [
+            science_input.get_time_range()[0]
+            for science_input in primary_science_inputs
+        ]
+        cadence_start_date = sorted(start_dates)[0]
```

Comment:
Can this be written as `min(...)` rather than needing to sort and then index into the 0th place.